be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp

```diff
@@ -813,6 +920,18 @@ Status VerticalSegmentWriter::write_batch() {
         _num_rows_written += data.num_rows;
     }
 
+    if (_opts.write_type == DataWriteType::TYPE_DIRECT ||
+        _opts.write_type == DataWriteType::TYPE_SCHEMA_CHANGE) {
+        for (auto& data : _batched_blocks) {
+            RETURN_IF_ERROR(_append_block_with_variant_subcolumns(data));
+        }
+    }
+
+    for (auto& column_writer : _column_writers) {
```

Comment:
why move to here?

Reply:
_append_block_with_variant_subcolumns may add new column writers